pkg/awsutils/awssession/session.go

```diff
@@ -0,0 +1,78 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package awssession
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/aws/amazon-vpc-cni-k8s/pkg/utils/logger"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"strconv"
+	"time"
+)
+
+// Http client timeout env for sessions
+const httpTimeoutEnv = "HTTP_TIMEOUT"
+
+var (
+	version string
+	log = logger.Get()
+	// HTTP timeout default value in seconds (10 seconds)
+	httpTimeoutValue = 10 * time.Second
+)
+
+func httpTimeout() time.Duration {
+	httpTimeoutEnvInput := os.Getenv(httpTimeoutEnv)
+	// if httpTimeout is not empty, we convert value to int and overwrite default httpTimeoutValue
+	if httpTimeoutEnvInput != "" {
+		if input, err := strconv.Atoi(httpTimeoutEnvInput); err == nil && input >= 1 {
```

Comment:
Should we maybe cap the input to be >= 10 seconds (default value)?

Reply:
Just curious...Actually, why would anyone prefer to reduce the default timeout value? What will it address? I mean, if anything one would probably want to increase the default timeout value? Should we instead, set the default 10s as the min limit and set an arbitrary max limit value?

Reply:
Yeah, thats why I was recommending we have input >= 10 which will set the min to 10 seconds. We can also have max based on the suggested max value from EC2.